src/main/java/org/tron/common/runtime/RuntimeImpl.java

```diff
@@ -270,25 +266,82 @@ private long getEnergyLimit(AccountCapsule creator, AccountCapsule caller,
     }
   }
 
-  private double getThisTxCPULimitInUsRatio() {
+  public long getEnergyLimitWithFixRatio(AccountCapsule creator, AccountCapsule caller,
+      TriggerSmartContract contract, long feeLimit, long callValue) {
+
+    long callerEnergyLimit = getEnergyLimitWithFixRatio(caller, feeLimit, callValue);
+    if (Arrays.equals(creator.getAddress().toByteArray(), caller.getAddress().toByteArray())) {
+      // when the creator calls his own contract, this logic will be used.
+      // so, the creator must use a BIG feeLimit to call his own contract,
+      // which will cost the feeLimit TRX when the creator's frozen energy is 0.
+      return callerEnergyLimit;
+    }
+
+    long creatorEnergyLimit = 0;
+    ContractCapsule contractCapsule = this.deposit
+        .getContract(contract.getContractAddress().toByteArray());
+    long consumeUserResourcePercent = contractCapsule.getConsumeUserResourcePercent();
+
+    long originEnergyLimit = contractCapsule.getOriginEnergyLimit();
+
+    if (consumeUserResourcePercent <= 0) {
+      creatorEnergyLimit = min(energyProcessor.getAccountLeftEnergyFromFreeze(creator),
+          originEnergyLimit);
+    } else if (consumeUserResourcePercent < Constant.ONE_HUNDRED) {
+      // creatorEnergyLimit =
+      // min(callerEnergyLimit * (100 - percent) / percent, creatorLeftFrozenEnergy, originEnergyLimit)
+
+      creatorEnergyLimit = min(
+          BigInteger.valueOf(callerEnergyLimit)
+              .multiply(BigInteger.valueOf(Constant.ONE_HUNDRED - consumeUserResourcePercent))
+              .divide(BigInteger.valueOf(consumeUserResourcePercent)).longValueExact(),
+          min(energyProcessor.getAccountLeftEnergyFromFreeze(creator), originEnergyLimit)
+      );
+    } else {
+    }
+    return Math.addExact(callerEnergyLimit, creatorEnergyLimit);
+  }
+
+  public long getEnergyLimit(AccountCapsule creator, AccountCapsule caller,
+      TriggerSmartContract contract, long feeLimit, long callValue) {
+    if (isNewVersion()) { // TODO according to version
+      return getEnergyLimitWithFixRatio(creator, caller, contract, feeLimit, callValue);
+    } else {
+      return getEnergyLimitWithFloatRatio(creator, caller, contract, feeLimit, callValue);
+    }
+  }
+
+  public long getEnergyLimit(AccountCapsule creator, long feeLimit, long callValue) {
+    if (isNewVersion()) {  // TODO according to version
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Change this condition so that it does not always evaluate to "false" ([squid:S2583](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2583))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1670&issues=AWbSdHUuSgN4uCclEWnG&open=AWbSdHUuSgN4uCclEWnG)

[ISSUE_KEY]: AWbSdHUuSgN4uCclEWnG